### PR TITLE
feat(524): add feature test for api token

### PR DESCRIPTION
**Pull request by FenrirUnbound**
## Context

As an alternative to passwords, Screwdriver could allow access tokens to be exchanged for JWTs. This would enable another authentication workflow besides the standard web application flow. 

## Objective

A feature file to represent the functionality behind #524 . 

## Miscellaneous

* Relates to #532

### CODE REVIEW

**Comment by d2lam** [comment]
Let's fix the pipeline first.

File: features/api-token.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    A user API token can be created and used in place of a password when
+    interacting with the API. This would allow users to obtain valid JWTs and
+    interact with the Screwdriver API in a programmatic way.
+
+    Rules:
+        - Tokens should be specifically tied to the user that issued them
+        - Tokens should have a label/name associated with them
+        - Users can add a description to their issued tokens, so they know what they're using them for
+        - User can see when a token was last used
+        - Users can revoke a token at any time.
```
Comment (r3rastogi) [inline features/api-token.feature]: Please remove the period at the end of this line; it's the only bullet with closing punctuation

File: features/api-token.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        - Users can add a description to their issued tokens, so they know what they're using them for
+        - User can see when a token was last used
+        - Users can revoke a token at any time
+        - No one can ever look up the raw value after initially generating the token
+
+    Background:
+        Given the user "calvin" exists
+
+    @ignore
+    Scenario: New API Token
```
Comment (minz1027) [inline features/api-token.feature]: `Add` or `Generate New API Token`? Just to be consistent with other scenarios~ They all start with verb

Comment (FenrirUnbound) [inline features/api-token.feature]: Amended

**Pull request merged**